lib/improver/tests/spotdata/spotdata/test_extrema.py

```diff
@@ -0,0 +1,328 @@
+from improver.spotdata.extrema import make_local_time_cube
+from improver.spotdata.extrema import get_valid_dates
+
+
+class Test_extrema(IrisTest):
+    """Setup tests for plugin and functions in extrema.py"""
+
+    def setUp(self):
+        """Set up the initial conditions for tests."""
+
+        times = range(414048, 414097, 1)  # Hours since 00Z-01-01-1970
+        n_times, n_data = len(times), 27
+        indices = DimCoord(np.arange(n_data), long_name='index',
+                           units='1')
+        time_units = cf_units.Unit('hours since 1970-01-01 00:00:00',
+                                   calendar='gregorian')
+        time_coord = DimCoord(times, standard_name='time',
+                              units=time_units)
+        # UTC offset arranged to sequence -12 --> 14.
+        utc_offsets = range(-12, 15) * int(n_data/24)
+        utc_offset = AuxCoord(utc_offsets, long_name='utc_offset',
+                              units='hours')
+        # Data arranged to ascend 0 --> n_data, so each site shows a constant
+        # temperature over time; this will need to be modified to check
+        # analysis collapse method.
+        data = np.array([range(n_data)*n_times])
+        data.resize(n_times, n_data)
+
+        cube = Cube(data,
+                    long_name="air_temperature",
+                    dim_coords_and_dims=[(time_coord, 0), (indices, 1)],
+                    aux_coords_and_dims=[(utc_offset, 1)],
+                    units="K")
+        self.cube = cube
+        self.data = data
+        self.time_coord = time_coord
+        self.n_data = n_data
+
+
+class Test_ExtractExtrema(Test_extrema):
+    """
+    Test the extraction of maximma/minima values in given periods, where the
+    periods are in local time.
+
+    Imagine that 27 sites sample all timeszones from UTC-12 to UTC+14.
+
+    """
+
+    def test_time_coordinates_24_hour(self):
+        """
+        Time coordinate should be a series of mid points calculated from the
+        start hour + half the period. Each should have an associated pair of
+        bounds that show the range over which the extrema values have been
+        calculated.
+
+        The first day is the first day for which any site has valid data. The
+        UTC time coord starts at 00 UTC on 27th April 2017, so the first day
+        in which any data falls is 26th April 2017 (for any UTC-N sites).
+        The first 24 hours starting at 00 therefore runs 00 26th to 00 27th.
+        Subsequent 24 hour periods are then expected to the latest day for
+        which any site has data; a UTC+14 site.
+
+        Input data spans 48 hours, this will spread to three days with timezone
+        adjustments.
+
+        """
+        n_periods = 72/24
+        mid_start = mktime(dt(2017, 03, 26, 12).utctimetuple())/3600.
+        lower_bound = mktime(dt(2017, 03, 26, 00).utctimetuple())/3600.
+        upper_bound = mktime(dt(2017, 03, 27, 00).utctimetuple())/3600.
```

Comment:
Github highlights these leading-zero-integers in red because some systems interpret them as octal values. It would be better not to use the leading zero.